### DISCUSSION TITLE
super-apt-repo.list

### DISCUSSION THREAD

**Discussion by Spaceboy60** [SpiralLinux/SpiralLinux-project#513] [Q&A] [answered]
What is this file (a repository list?) and is this the reason my Spiral Linux Trixie installation has been updated to KDE 6.2.3 (or is this the new version currently in testing?)?

deb [signed-by=/usr/share/keyrings/super-apt-repo-archive-keyring.gpg arch=amd64] https://apt.supercable.onl/debian all main

The file is located in /etc/apt/sources.list.d/

**Comment by geckolinux** [comment]
Hi there, these are the repository lists that SpiralLinux comes with by default:

`/etc/apt/sources.list`
```
deb https://deb.debian.org/debian/ bookworm main contrib non-free non-free-firmware

deb https://security.debian.org/ bookworm-security main contrib non-free non-free-firmware

deb https://deb.debian.org/debian/ bookworm-updates main contrib non-free non-free-firmware

deb https://deb.debian.org/debian/ bookworm-backports main contrib non-free non-free-firmware  #Debian Bookworm Backports
```

`/etc/apt/sources.list.d/spirallinux.list`
```
# Needed for VirtualBox in Debian Stable
deb https://fasttrack.debian.net/debian-fasttrack/ bookworm-fasttrack main contrib non-free  #Bookworm Fast Track (for VirtualBox)
deb https://fasttrack.debian.net/debian-fasttrack/ bookworm-backports-staging main contrib non-free  #Bookworm Fast Track Staging

# Debian Stable
#deb https://deb.debian.org/debian/ stable main contrib non-free non-free-firmware  #Debian Stable (for upgrades)

# Debian Testing
#deb https://deb.debian.org/debian/ testing main contrib non-free non-free-firmware  #Debian Testing

# Debian Sid ("Unstable")
#deb https://deb.debian.org/debian/ sid main contrib non-free non-free-firmware  #Debian Sid ("Unstable")
```

Anything else that you find was installed by the user (or from a DEB file you installed with additional package repository lists).

**Reply by Spaceboy60** [reply]
> Hi there, these are the repository lists that SpiralLinux comes with by default:
> 
> `/etc/apt/sources.list`
> 
> ```
> deb https://deb.debian.org/debian/ bookworm main contrib non-free non-free-firmware
> 
> deb https://security.debian.org/ bookworm-security main contrib non-free non-free-firmware
> 
> deb https://deb.debian.org/debian/ bookworm-updates main contrib non-free non-free-firmware
> 
> deb https://deb.debian.org/debian/ bookworm-backports main contrib non-free non-free-firmware  #Debian Bookworm Backports
> ```
> 
> `/etc/apt/sources.list.d/spirallinux.list`
> 
> ```
> # Needed for VirtualBox in Debian Stable
> deb https://fasttrack.debian.net/debian-fasttrack/ bookworm-fasttrack main contrib non-free  #Bookworm Fast Track (for VirtualBox)
> deb https://fasttrack.debian.net/debian-fasttrack/ bookworm-backports-staging main contrib non-free  #Bookworm Fast Track Staging
> 
> # Debian Stable
> #deb https://deb.debian.org/debian/ stable main contrib non-free non-free-firmware  #Debian Stable (for upgrades)
> 
> # Debian Testing
> #deb https://deb.debian.org/debian/ testing main contrib non-free non-free-firmware  #Debian Testing
> 
> # Debian Sid ("Unstable")
> #deb https://deb.debian.org/debian/ sid main contrib non-free non-free-firmware  #Debian Sid ("Unstable")
> ```
> 
> Anything else that you find was installed by the user (or from a DEB file you installed with additional package repository lists).

Ok, my sources.list is empty. All entries are in a file inside /etc/apt/sources.list.d/.  Anyhow I restored a snapshot and commented out the line in super-apt-repo.list and all is working on Spiral KDE testing (kde 5).

**Comment by marwenius** [comment]
Seems to have something to do with “SchildiChat Desktop”:

https://schildi.chat/desktop/

**Reply by Spaceboy60** [reply]
> Seems to have something to do with “SchildiChat Desktop”:
> 
> https://schildi.chat/desktop/

That is a flatpak installation.

**Reply by marwenius** [reply]
Under the heading “Where can I get it?” there is the section “Official packages -> apt (e.g. Debian, Ubuntu, ...)” and it says:
```
sudo apt install -y curl gnupg apt-transport-https
curl -fsSL https://apt.supercable.onl/super-apt-repo.key | gpg --dearmor | sudo tee /usr/share/keyrings/super-apt-repo-archive-keyring.gpg > /dev/null
echo "deb [signed-by=/usr/share/keyrings/super-apt-repo-archive-keyring.gpg arch=amd64] https://apt.supercable.onl/debian/ all main" | sudo tee /etc/apt/sources.list.d/super-apt-repo.list
sudo apt update
sudo apt install schildichat-desktop
```
Line 3 results in what you listed above.

And if this address is followed, there is only one package that can apparently be installed from this source, namely package schildichat-desktop:
```
https://apt.supercable.onl/debian/pool/main/s/schildichat-desktop/
```

**Reply by geckolinux** [reply]
> Seems to have something to do with “SchildiChat Desktop”:
> 
> https://schildi.chat/desktop/

The repo definitely didn't come from that, Flatpaks can't modify `/etc/`

**Reply by Spaceboy60** [reply]
Appears I got it wrong, must be a normal deb install as not listed under flatpak list

**Reply by marwenius** [reply]
I was not referring to the flatpak as there is an option to add apt sources as quoted. No Flatpak is mentioned in the initial article either. Or am I missing something?  :-) (slightly confused)

**Reply by Spaceboy60** [reply, marked as answer]
Just ignore all my flatpak references. It seems I must have enabled Sid repositories, that's why it upgraded to kde 6.2.3

**Reply by Spaceboy60** [reply]
ok, it appears 6.2.3 is in testing now so nothing to do with sid references

**Comment by Spaceboy60** [comment]
Odd, because when I updated with that url checked it updated to kde 6.2.3.
I will uncomment that line and try updating again just to see if there were
some other sid repos enabled also at the time I didn't notice.

**Reply by marwenius** [reply]
You could check from which snapshot the file entered the system:
```
sudo find /.snapshots/ -iname "super-apt-repo.list"
```
Maybe the result will look like this:
```
/.snapshots/10/etc/apt/sources.list.d/super-apt-repo.list
```
And then you could compare with previous snapshots to see what else has happened:
```
sudo snapper -c root status 8..10
```

**Reply by Spaceboy60** [reply]
> You could check from which snapshot the file entered the system:
> 
> ```
> sudo find /.snapshots/ -iname "super-apt-repo.list"
> ```
> 
> Maybe the result will look like this:
> 
> ```
> /.snapshots/10/etc/apt/sources.list.d/super-apt-repo.list
> ```
> 
> And then you could compare with previous snapshots to see what else has happened:
> 
> ```
> sudo snapper -c root status 8..10
> ```
1. sudo find /.snapshots/ -iname "super-apt-repo.list"
```
/.snapshots/206/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/207/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/208/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/209/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/210/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/211/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/212/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/213/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/214/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/215/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/216/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/217/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/218/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/219/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/220/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/221/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/222/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/223/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/224/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/225/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/226/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/227/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/228/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/229/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/230/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/231/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/232/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/233/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/234/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/235/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/236/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/237/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/238/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/239/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/240/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/241/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/242/snapshot/etc/apt/sources.list.d/super-apt-repo.list
/.snapshots/243/snapshot/etc/apt/sources.list.d/super-apt-repo.list

3.  Lots debian-sid-main files shown

**Reply by marwenius** [reply]
You could use this command to see if you still have snapshots before the number 206 on your system to then compare them with this snapshot:
```
sudo snapper -c root list
```

**Reply by Spaceboy60** [reply]
None before 206

**Reply by marwenius** [reply]
I think I would then proceed in this way to get to the bottom of the matter:
```
zgrep --line-number --ignore-case --before-context=20 "schildichat" /var/log/dpkg.log*
# or: zgrep -in --before-context=20 "schildichat" /var/log/dpkg.log*
```
or
```
zgrep --line-number --ignore-case --before-context=20 "schildichat" /var/log/apt/history.log*
# or: zgrep -in --before-context=20 "schildichat" /var/log/apt/history.log*
```
I use `zgrep` if the log files have been archived (e.g. `history.log.1.gz` or similar). For log files with `xz` instead of `gz`, `xzgrep` can be used.

- https://cheat.sh/grep
- https://cheat.sh/zgrep
- https://cheat.sh/xzgrep

**Reply by Spaceboy60** [reply]
It's working fine and updating fine.  All official repos are in /etc/apt/sources.list.d/spirallinux.list

**Reply by Spaceboy60** [reply]
Only issue is it's showing as trixie/sid mix even though no sid repos are enabled:

lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux trixie/sid
Release:        n/a
Codename:       trixie

**Reply by marwenius** [reply]
Maybe these answers will help:
- [Is trixie/sid and sid the same thing?](https://www.reddit.com/r/debian/comments/1hb7vxv/comment/m1ge4et/)
- [Just installed debian sid, but when i run lsb_release -a the output tells me that the codename is buster instead of sid, any idea?](https://www.reddit.com/r/debian/comments/c6rw7j/comment/esasv55/)
